src/tiled/mapobjectmodel.cpp

```diff
@@ -233,7 +242,9 @@ Qt::ItemFlags MapObjectModel::flags(const QModelIndex &index) const
     if (index.column() == 0)
         rc |= Qt::ItemIsUserCheckable | Qt::ItemIsEditable;
     else if (toMapObject(index))
-        rc |= Qt::ItemIsEditable; // MapObject type
+        if (index.column() == Type) { // allow to edit only type column
+            rc |= Qt::ItemIsEditable;
+        }
```

Comment:
Coding style: Please wrap the outer if/else if with braces and remove them from the nested if.